common/random.cpp

```diff
@@ -46,7 +46,7 @@ void RandomSource::setSeed(uint32 seed) {
 uint RandomSource::getRandomNumber(uint max) {
 	_randSeed = 0xDEADBF03 * (_randSeed + 1);
 	_randSeed = (_randSeed >> 13) | (_randSeed << 19);
-	return _randSeed % (max + 1);
+	return _randSeed % MAX(max, 1u);
```

Comment:
That does not look quite right since with this change we now get a result in the range [0, max - 1] instead of [0, max].

Reply:
Pushed a fixed version of the commit where the range is preserved (except on when max == UINT_MAX).

Reply:
This is better, but you still get the issue that you cannot get max in the case where max == UINT_MAX.

I think you will need something like this:
```c++
if (max == UINT_MAX)
    return _randSeed;
else
    return _randSeed % (max + 1);
```

Reply:
Thanks, I've pushed a fixed version.

Reply:
Nice catch!
The code looks good to me, after the latest commit.